src/mds/MDSRank.cc

```diff
@@ -901,9 +901,22 @@ Session *MDSRank::get_session(Message *m)
 {
   Session *session = static_cast<Session *>(m->get_connection()->get_priv());
   if (session) {
+    session->put(); // do not carry ref
     dout(20) << "get_session have " << session << " " << session->info.inst
 	     << " state " << session->get_state_name() << dendl;
-    session->put();  // not carry ref
+    // Check if we've imported an open session since
+    Session *imported_session = sessionmap.get_session(session->info.inst.name);
+    if (imported_session && imported_session != session) {
+      /* do not imported_session->put() because set_priv consumes ref */
```

Comment:
imported_session is from sessionmap.get_session(). it has no extra reference.

Reply:
Oh right, good catch!